frontend/src/lib/components/heatmaps/HeatMapsSettings.tsx

```diff
@@ -0,0 +1,235 @@
+import { IconInfo } from '@posthog/icons'
+import { useValues } from 'kea'
+import { HeatmapFilters, HeatmapFixedPositionMode } from 'lib/components/heatmaps/types'
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { LemonLabel } from 'lib/lemon-ui/LemonLabel'
+import { LemonSegmentedButton } from 'lib/lemon-ui/LemonSegmentedButton'
+import { LemonSelect } from 'lib/lemon-ui/LemonSelect'
+import { LemonSlider } from 'lib/lemon-ui/LemonSlider'
+import React, { useState } from 'react'
+
+import { HEATMAP_COLOR_PALETTE_OPTIONS, heatmapLogic } from '~/toolbar/elements/heatmapLogic'
+
+const ScrollDepthJSWarning = (): JSX.Element | null => {
+    const { scrollDepthPosthogJsError } = useValues(heatmapLogic)
+
+    if (!scrollDepthPosthogJsError) {
+        return null
+    }
+
+    return (
+        <p className="my-2 bg-danger-highlight border border-danger rounded p-2">
+            {scrollDepthPosthogJsError === 'version' ? (
+                <>This feature requires a newer version of posthog-js</>
+            ) : scrollDepthPosthogJsError === 'disabled' ? (
+                <>
+                    Your posthog-js config has <i>disable_scroll_properties</i> set - these properties are required for
+                    scroll depth calculations to work.
+                </>
+            ) : null}
+        </p>
+    )
+}
+const SectionSetting = ({
+    children,
+    title,
+    info,
+}: {
+    children: React.ReactNode
+    title: React.ReactNode
+    info?: React.ReactNode
+}): JSX.Element => {
+    const [showInfo, setShowInfo] = useState(false)
+    return (
+        <div className="space-y-2 mb-2">
+            <div className="flex items-center gap-2">
+                <LemonLabel className="flex-1">
+                    {title}
```

Comment:
Might be better to explicitly type this as a string and then wrap it to maintain a better dom structure
```suggestion
                    <span>{title}</span>
```